GliaWidgets/ViewModel/Chat/Unread/UnreadMessagesHandler.swift

```diff
@@ -2,34 +2,47 @@ class UnreadMessagesHandler {
     private let unreadMessages: ObservableValue<Int>
     private let isWindowVisible: ObservableValue<Bool>
     private let isViewVisible: ObservableValue<Bool>
+    private let isChatScrolledDown: ObservableValue<Bool>
 
-    init(unreadMessages: ObservableValue<Int>,
-         isWindowVisible: ObservableValue<Bool>,
-         isViewVisible: ObservableValue<Bool>) {
+    init(
+        unreadMessages: ObservableValue<Int>,
+        isWindowVisible: ObservableValue<Bool>,
+        isViewVisible: ObservableValue<Bool>,
+        isChatScrolledToBottom: ObservableValue<Bool>
+    ) {
         self.unreadMessages = unreadMessages
         self.isWindowVisible = isWindowVisible
         self.isViewVisible = isViewVisible
+        self.isChatScrolledDown = isChatScrolledToBottom
         isWindowVisible.addObserver(self) { _, _ in
             self.checkVisible()
         }
         isViewVisible.addObserver(self) { _, _ in
             self.checkVisible()
         }
+        isChatScrolledDown.addObserver(self) { _, _ in
+            self.checkVisible()
```

Comment:
I'd like to propose to avoid memory leaks and use weak references to self here and on lines 18, 21.